provider/pkg/gen/schema.go

```diff
@@ -1024,9 +1024,9 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 		if !isOutput && readOnly {
 			continue
 		}
-		if isOutput && name == "id" {
-			continue
-		}
+		//if isOutput && name == "id" {
+		//	continue
+		//}
```

Comment:
As briefly discussed in slack, I think we should keep doing this for resources but not types

Reply:
Thanks for the insight @mikhailshilkov. I have updated the PR now. Could you take another look?